Version 2.0.0

mciuchitu opened the pull request:
<!--
  Thanks for creating a Pull Request! Before you submit, please make sure
  you've done the following:

  - I read the contributing document at https://github.com/goparrot/geocoder/blob/master/CONTRIBUTING.md#contributing
-->

<!--
  Make our lives easier! Choose one of the following by uncommenting it:
-->

<!-- This is a 🐛 bug fix. -->

This is a 🙋 feature or enhancement.

<!-- This is a 🔦 documentation change. -->

<!--
  Before you submit this pull request, make sure to have a look at the following
  checklist. If you don't know how to do some of these, that's fine! Submit
  your pull request and we will help you out on the way.

  - I've added tests (if it's a bug, feature or enhancement)
  - I've adjusted the documentation (if it's a feature or enhancement)
  - The test suite passes locally (run `npm test` to verify this)
-->

## Summary

<!--
  Provide a description of what your pull request changes.
-->

Make `EventBus#pubish` method awaitable

## Context

<!--
  Is this related to any GitHub issue(s)?

  You can use keywords to automatically close the related issue.
  For example, (all of) the following will close issue #4567 when your PR is merged.

  Closes #4567
  Fixes #4567
  Resolves #4567

  Use any one of the above as applicable.
-->

ruscon commented on package.json:
```diff
@@ -79,15 +79,15 @@
         "@nestjs/core": "^7.1.3",
         "@nestjs/cqrs": "^7.0.0",
-        "@types/amqp-connection-manager": "^2.0.10",
+        "@types/amqp-connection-manager": "^2.0.12",
```
remove, `amqp-connection-manager` already has its own typing

Commit:
refactor(event-bus): make publish methods async

BREAKING CHANGE: EventBus `publish` and `publishAll` methods now return promises that should be
properly handled

Commit:
refactor(event): do not publish event locally by default

BREAKING CHANGE: Pub-sub events are no more locally published by default. To publish the event both
to RabbitMQ and locally use the `withLocal` event configuration method

Commit:
feat(event-handler): catch and log unhandled exceptions

Commit:
feat(event): make message field private

Commit:
refactor(event): replace payload method with readonly field

BREAKING CHANGE: Event paylod is no longer a method, but a field. It should not be invoked to
retrieve its content

Commit:
refactor: improve typings

Commit:
chore: use goparrot eslint config

Commit:
chore: update husky

Commit:
feat: move exchange to the event decorator

BREAKING CHANGE: Refer to Migration Guide from 1.5.2 to 2.0.0

Commit:
docs: add a migration guide to the 2.0.0 version

Commit:
chore(release): 2.0.0
